### PR TITLE
Fix #17070, move docs out of HelpDB too.

### LINKED ISSUE #17070: Keyword argument docs for read are misleading
This is via @aviks, we were looking at this in the morning:
```
help?> read
search: read read! readdlm readdir readcsv readall readlink readline readuntil readlines readchomp

  read(stream::IO, T)

  Read a single value of type T from stream, in canonical binary representation.

  read(stream::IO, T, dims)

  Read a series of values of type T from stream, in canonical binary representation. dims is either
  a tuple or a series of integer arguments specifying the size of the Array{T} to return.

  read(filename::AbstractString, args...)

  Open a file and read its contents. args is passed to read: this is equivalent to open(io->read(io,
  args...), filename).

  read(stream::IO, nb=typemax(Int); all=true)

  Read at most nb bytes from stream, returning a Vector{UInt8} of the bytes read.

  If all is true (the default), this function will block repeatedly trying to read all requested
  bytes, until an error or end-of-file occurs. If all is false, at most one read call is performed,
  and the amount of data returned is device-dependent. Note that not all stream types support the
  all option.

julia> s = TCPSocket()
TCPSocket(init, 0 bytes waiting)

julia> read(s, 10; all=true)
ERROR: MethodError: no method matching (::Base.#kw##read)(::Array{Any,1}, ::Base.#read, ::TCPSocket, ::Int64)
Closest candidates are:
  #read(::Array{Any,1}, ::Base.#read, ::IOStream, ::Integer)
 in eval(::Module, ::Any) at .\boot.jl:231
 in macro expansion at .\REPL.jl:92 [inlined]
 in (::Base.REPL.##1#2{Base.REPL.REPLBackend})() at .\event.jl:46
```

The `all` keyword argument is actually only implemented for `IOStream`, not all subtypes of `IO`.

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

**Comment by tkelman** [comment]
should mention what #17070 means in the commit message

File: doc/stdlib/io-network.rst [hunk 1]
```diff
@@ -168,11 +168,11 @@ General I/O
 
    See ``read`` for a description of the ``all`` option.
 
-.. function:: read(stream::IO, nb=typemax(Int); all=true)
+.. function:: read(s::IOStream, nb::Integer; all=true)
```
Comment (tkelman) [inline doc/stdlib/io-network.rst]: later in `base/io.jl` there is
```
# read up to nb bytes from s, returning a Vector{UInt8} of bytes read.
function read(s::IO, nb=typemax(Int))
```
that still has the default value of `nb`

Comment (kshyatt) [inline doc/stdlib/io-network.rst]: It doesn't have the `all` kwarg though. You can't do `read(s; all=false)`, can you?

Comment (tkelman) [inline doc/stdlib/io-network.rst]: you can't, so maybe the signature with `nb=typemax(Int)` default value but no kwarg should be a separate docstring?

Comment (kshyatt) [inline doc/stdlib/io-network.rst]: I agree, will add.

**Comment by kshyatt** [comment]
Travis fail isn't my fault, right?

File: base/iostream.jl [hunk 2]
```diff
@@ -246,6 +246,16 @@ function read(s::IOStream)
...
 
+"""
+    read(s::IOStream, nb::Integer; all=true)
+
+Read at most `nb` bytes from `s`, returning a `Vector{UInt8}` of the bytes read.
+
+If `all` is `true` (the default), this function will block repeatedly trying to read all
+requested bytes, until an error or end-of-file occurs. If `all` is `false`, at most one
+`read` call is performed, and the amount of data returned is device-dependent. Note that not
+all stream types support the `all` option.
```
Comment (tkelman) [inline base/iostream.jl]: needs another make docs run

**Comment by tkelman** [comment]
timeout, probably related to the perf regression

**Pull request merged**